### DISCUSSION TITLE
Machine Max Extents

### DISCUSSION THREAD

**Discussion by MDeBaise1** [cncjs/cncjs#786] [General]
I'm having trouble trying to set my machine extents. I'm trying to set the machine up so I can set each corner as the max X left and right and max Y front and back. This might be beyond what I should be asking in this form and if it is please just let me know. I want to set these max extents so (1) I know how large I can cut and (2) where the limits are. I must be missing something because I feel this is a basic setup feature. 

Any thoughts would be great!

**Comment by MitchBradley** [comment]
There are three different ways to construe the question.

Setting the extents so the machine will refuse to go outside the limits is typically done in the controller firmware, not in CNCjs.  If the controller runs GRBL, Google "GRBL soft limits".

CNCjs can display - but not enforce - the machine limits.  That is done via the setting panel which is accessed via the Gears icon on the left side - choose the Machine Profiles tab and add a new profile.  Go back to the main workspace (axis icon on left) and select your newly created profile below the 3D visualizer

Many CAM programs can enforce limits by setting up a machine definition.  How to do so depends on the particular CAM program.

**Reply by MDeBaise1** [reply]
Ahhh.... I will look into that. THANK YOU!
